sqllin-driver/src/nativeMain/kotlin/com/ctrip/sqllin/driver/cinterop/NativeDatabase.kt

```diff
@@ -71,10 +70,10 @@ internal class NativeDatabase private constructor(val dbPointer: CPointer<sqlite
             }
 
             // Check that the database is really read/write when that is what we asked for.
-            if ((sqliteFlags and SQLITE_OPEN_READWRITE > 0) && sqlite3_db_readonly(db, null) != 0) {
+            /*if ((sqliteFlags and SQLITE_OPEN_READWRITE > 0) && sqlite3_db_readonly(db, null) != 0) {
                 sqlite3_close_v2(db)
                 throw sqliteException("Could not open the database in read/write mode")
-            }
+            }*/
```

Comment:
shouldn't you keep this in but change the if to `if (!configuration.isReadOnly && sqlite3_db_readonly(db, null) != 0)`

Reply:
but also I dont think this is possible anyway because it would fail [here](https://github.com/ctripcorp/SQLlin/pull/52/commits/160903df92169e2c6727f810a5c12102389c8f95#diff-1385c50e4a06f1f0c13359b4ef9f0a217d12d12cb73842dbc13addee6bc28077R57)